modular_zubbers/code/modules/credits/credits.dm

```diff
@@ -0,0 +1,193 @@
+#define CREDIT_ROLL_SPEED 100
+#define CREDIT_SPAWN_SPEED 20
+#define CREDIT_ANIMATE_HEIGHT (14 * world.icon_size)
+#define CREDIT_EASE_DURATION 22
+GLOBAL_VAR_INIT(end_credits_song, null)
+GLOBAL_VAR_INIT(end_credits_title, null)
+GLOBAL_LIST(end_titles)
+#define JOINTEXT(X) jointext(X, null)
+/datum/controller/subsystem/ticker/declare_completion()
+	for(var/client/C)
+		if(!C.credits)
+			C.RollCredits()
+	for(var/thing in GLOB.clients)
+		var/client/C = thing
+		if (!C)
+			continue
+	. = ..()
+
+/datum/controller/subsystem/ticker/show_roundend_report(client/C, report_type = null)
+	. = ..()
+	for(var/iteration in GLOB.clients)
+		var/client/clients = iteration
+		if(!clients.credits)
+			clients.RollCredits()
+/* /client
+	var/list/credits */
+
+
+/client/proc/RollCredits()
+	set waitfor = FALSE
+
+
+	if(!GLOB.end_titles)
+		GLOB.end_titles = generate_titles()
+
+	LAZYINITLIST(credits)
+
+/* 	if(mob)
+		mob.overlay_fullscreen("meeting",/atom/movable/screen/fullscreen/emergency_meeting) */
+//	sound_to_playing_players('modular_skyrat/master_files/sound/music/elibao.ogg', volume = 100, vary = FALSE, frequency = 0, channel = 0, pressure_affected = FALSE)
+
+	var/list/_credits = credits
+	verbs += /client/proc/ClearCredits
+	for(var/I in GLOB.end_titles)
+		if(!credits)
+			return
+		var/atom/movable/screen/credit/T = new(null, I, src)
+		_credits += T
+		T.rollem()
+		sleep(CREDIT_SPAWN_SPEED)
+	sleep(CREDIT_ROLL_SPEED - CREDIT_SPAWN_SPEED)
+
+	ClearCredits()
+	verbs -= /client/proc/ClearCredits
+
+/client/proc/ClearCredits()
+	set name = "Stop End Titles"
+	set category = "OOC"
+	verbs -= /client/proc/ClearCredits
+	QDEL_NULL(credits)
```

Comment:
This proc may be invoked twice- I don't know if that's an issue, I imagine not, but I'm noting it here anyways.